### PR TITLE
end to end tests for affinities 

### PR DESCRIPTION
**Pull request by preetapan**
Intended to run in internal CI/CD environment, builds on top of #5012 

This PR adds basic tests for affinities.

### CODE REVIEW

**Comment by preetapan** [comment]
closing this, will open another one (incorrect rebase)

File: e2e/affinities/affinities.go [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+
+func (tc *BasicAffinityTest) TestSingleAffinities(f *framework.F) {
+	nomadClient := tc.Nomad()
+
+	// Parse job
+	job, err := jobspec.ParseFile("affinities/input/aff1.nomad")
+	require := require.New(f.T())
+	require.Nil(err)
+	uuid := uuid.Generate()
+	jobId := helper.StringToPtr("aff" + uuid[0:8])
```
Comment (dadgar) [inline e2e/affinities/affinities.go]: Why truncate?

File: e2e/affinities/affinities.go [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+		},
+	})
+}
+
+func (tc *BasicAffinityTest) TestSingleAffinities(f *framework.F) {
+	nomadClient := tc.Nomad()
+
+	// Parse job
+	job, err := jobspec.ParseFile("affinities/input/aff1.nomad")
+	require := require.New(f.T())
```
Comment (dadgar) [inline e2e/affinities/affinities.go]: I have a small stylistic preference for this to be defined on the top of the func

File: e2e/affinities/affinities.go [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+
+	allocs, _, _ := jobs.Allocations(*job.ID, false, nil)
+
+	// Verify affinity score metadata
+	for _, allocStub := range allocs {
+		alloc, _, err := jobAllocs.Info(allocStub.ID, nil)
+		require.Nil(err)
+		require.NotEmpty(alloc.Metrics.ScoreMetaData)
+		// Expect node affinity score to be 1.0 (dc1 nodes) if found
+		// TODO(preetha) make this more realistic
```
Comment (dadgar) [inline e2e/affinities/affinities.go]: What do you have it mind?

File: e2e/affinities/affinities.go [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+		for _, sm := range alloc.Metrics.ScoreMetaData {
+			score, ok := sm.Scores["node-affinity"]
+			if ok {
```
Comment (dadgar) [inline e2e/affinities/affinities.go]: Should the test fail if !ok?

File: e2e/affinities/input/aff1.nomad [hunk 5]
```diff
@@ -0,0 +1,22 @@
+job "test1" {
```
Comment (dadgar) [inline e2e/affinities/input/aff1.nomad]: Would it make sense to have the file names match the tests in some way?

File: e2e/affinities/affinities.go [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+		}
+	}
+
+}
+
+func (tc *BasicAffinityTest) TestMultipleAffinities(f *framework.F) {
+	nomadClient := tc.Nomad()
+
+	// Parse job
+	job, err := jobspec.ParseFile("affinities/input/aff2.nomad")
```
Comment (dadgar) [inline e2e/affinities/affinities.go]: Is 85-111 extractable into a function and reused by all 3?

File: e2e/affinities/affinities.go [hunk 7]
```diff
@@ -0,0 +1,228 @@
...
+		dcMatch := node.Datacenter == "dc1"
+		rackMatch := node.Meta != nil && node.Meta["rack"] == "r1"
+
+		// Figure out expected node affinity score based on whether both affinities match or just one does
+		expectedNodeAffinityScore := 0.0
+		if dcMatch && rackMatch {
+			expectedNodeAffinityScore = 1.0
+		} else if dcMatch || rackMatch {
+			expectedNodeAffinityScore = 0.5
+		}
```
Comment (dadgar) [inline e2e/affinities/affinities.go]: Should there be an else case that fails?

Comment (dadgar) [reply e2e/affinities/affinities.go]: Same on last test

File: e2e/affinities/affinities.go (hunk 4 above)
Comment (preetapan) [reply e2e/affinities/affinities.go]: not every node in the score metadata will match the affinity, so this is correct.

**Pull request closed without merging**